### PR TITLE
Made sure that DataSourceOption displays value and fires onChange/onBlur events

### LINKED ISSUE #15682: Min time interval not displayed in V6.0.0
Please include this information:
### What Grafana version are you using?
Grafana v6.0.0 (34a9a62)
### What datasource are you using?
Elasticsearch
### What OS are you running grafana on?
Centos
### What did you do?
Just config some panel with `min time interval` and the change to any other tab then change back, the panel is still showing data with pre configured value but the `min time interval` box will be empty. 
### What was the expected result?
Min time interval displayed correctly.

wx20190227-143443 2x

### PR DESCRIPTION
**Pull request by hugohaggmark**
Fixes #15682

- Added onBlur on DataSourceOption props
- Changed type for onChange on DataSourceOption props
- Added value to input in DataSourceOption
- Connected value for DataSourceOption to state
- Made some small refactorings

### CODE REVIEW

File: public/app/features/dashboard/panel_editor/DataSourceOption.tsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+  onBlur: (event: ChangeEvent<HTMLInputElement>) => void;
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
   tooltipInfo?: any;
 }
 
-export const DataSourceOptions: FC<Props> = ({ label, placeholder, name, value, onChange, tooltipInfo }) => {
+export const DataSourceOption: FC<Props> = ({ label, placeholder, name, value, onBlur, onChange, tooltipInfo }) => {
   return (
     <div className="gf-form gf-form--flex-end">
       <FormLabel tooltip={tooltipInfo}>{label}</FormLabel>
```
Comment (hugohaggmark) [inline public/app/features/dashboard/panel_editor/DataSourceOption.tsx]: Should we use Grafanas Input instead?

Comment (jschill) [reply public/app/features/dashboard/panel_editor/DataSourceOption.tsx]: Good idea! We can probably use the same as the Gauge? https://github.com/grafana/grafana/blob/590450291af2c4c15593923115f8159a786cef7c/public/app/plugins/panel/gauge/GaugeOptionsBox.tsx#L31

Comment (hugohaggmark) [reply public/app/features/dashboard/panel_editor/DataSourceOption.tsx]: Created issue for this #15785

**Review by jschill — COMMENTED** [review]
Looks good!

**Pull request merged**